src/Symfony/Component/RateLimiter/Policy/SlidingWindow.php

```diff
@@ -22,32 +22,12 @@
  */
 final class SlidingWindow implements LimiterStateInterface
 {
-    private $id;
-
-    /**
-     * @var int
-     */
-    private $hitCount = 0;
-
-    /**
-     * @var int
-     */
-    private $hitCountForLastWindow = 0;
-
-    /**
-     * @var int how long a time frame is
-     */
-    private $intervalInSeconds;
-
-    /**
-     * @var int the unix timestamp when the current window ends
-     */
-    private $windowEndAt;
-
-    /**
-     * @var bool true if this window has been cached
-     */
-    private $cached = true;
```

Comment:
this is true but the constructor sets it to false. this is very odd.

Reply:
Indeed. 🤔

Reply:
`$cached` becomes true once the window instance has been serialized for the first time. See `__sleep()`.